clojure-mode.el

```diff
@@ -997,8 +997,8 @@ nil."
 PATH is expected to be an absolute file path.
 
 If PATH is nil, use the path to the file backing the current buffer."
-  (let* ((relative (clojure-project-relative-path
-                    (or path (file-truename (buffer-file-name)))))
+  (let* ((path (or path (file-truename (buffer-file-name))))
+         (relative (clojure-project-relative-path path))
          (sans-file-type (substring relative 0 (- (length (file-name-extension path t)))))
```

Comment:
If you can't spot the problem, @bbatsov, `path` will be `nil` on this line when the function is called without any arguments.